src/components/discover-items/DiscoverItems.jsx

```diff
@@ -12,45 +15,93 @@ const DiscoverItems = (props) => {
   const [items, setItems] = useState([]);
   const [hasMore, setHasMore] = useState(true);
   const [pages, setPages] = useState(-1);
+  const [selectedCategory, setSelectedCategory] = useState(null);
+  const [categoryOptions, setCategoryOptions] = useState([]);
+  const [viewTradable, setViewTradable] = useState(false);
 
   const loadItems = useCallback(async () => {
     const nextPage = pages + 1;
     try {
       if (!hasMore) {
         return;
       }
-      const retrievedItems = await getDiscoverItems(nextPage * LIMIT, LIMIT);
+      const retrievedItems = await getDiscoverItems(nextPage * LIMIT, LIMIT, selectedCategory, viewTradable);
       const updatedHasMore = retrievedItems && retrievedItems.length >= LIMIT;
       setHasMore(updatedHasMore);
       setItems([...items, ...retrievedItems]);
       setPages(nextPage);
     } catch (e) {
       setToast({ message: "Unable to load items. Please try again later.", color: "danger" });
     }
-  }, [hasMore, items, pages]);
+  }, [hasMore, items, pages, selectedCategory, setToast, viewTradable]);
 
   const loadInitialItems = useCallback(async () => {
     const nextPage = 0;
     try {
-      const retrievedItems = await getDiscoverItems(nextPage * LIMIT, LIMIT);
+      const retrievedItems = await getDiscoverItems(nextPage * LIMIT, LIMIT, selectedCategory, viewTradable);
       const updatedHasMore = retrievedItems && retrievedItems.length >= LIMIT;
       setHasMore(updatedHasMore);
       setItems(retrievedItems);
       setPages(nextPage);
     } catch (e) {
       setToast({ message: "Unable to load items. Please try again later.", color: "danger" });
     }
-  }, []);
+  }, [selectedCategory, setToast, viewTradable]);
 
   const fetchNextPage = () => {
     loadItems();
   };
 
   useEffect(() => {
+    const loadCategoryOptions = async () => {
+      // setLoading(true);
+      try {
+        const options = await getCategories();
+        setCategoryOptions(options);
+      } catch (e) {
+        console.log(e);
+      } finally {
+        // setLoading(false);
+      }
+    };
+
+    loadCategoryOptions();
```

Comment:
For fetching + loading categories, should we have a separate useEffect that only runs once? Since category is not user-editable and it will be wasteful to keep querying for every discover page reload